Common/src/main/java/potionstudios/byg/common/entity/pumpkinwarden/PumpkinWarden.java

```diff
@@ -254,19 +254,50 @@ public BlockState getCarriedBlock() {
     static class PumpkinWardenTakeBlockGoal extends MoveToBlockGoal {
         private final PumpkinWarden warden;
 
+        /**
+         * The delay until {@link PumpkinWardenTakeBlockGoal#findNearestBlock()}
+         * attempts to find for a new melon or pumpkin block nearby.
+         * This delay is counted in ticks and is continuously decremented until it hits
+         * 0, at which point a new nearest block will be searched for.
+         * The value of this field will be -1 if the goal has already found a
+         * valid target block.
+         *
+         * <p>This field mainly exists for performance reasons.
+         */
+        private int searchNearestBlockDelay = 0;
+
         public PumpkinWardenTakeBlockGoal(PumpkinWarden p, double speed, int range, int y) {
             super(p, speed, range, y);
             this.warden = p;
         }
 
         @Override
         public boolean canUse() {
+            if (this.searchNearestBlockDelay > 0) {
+                this.searchNearestBlockDelay--;
+            }
             if (this.warden.getCarriedBlock() != null) {
                 return false;
             }
             return super.canUse();
         }
 
+        @Override
+        protected boolean findNearestBlock() {
+            if (this.searchNearestBlockDelay == -1 && this.isValidTarget(this.warden.level, this.blockPos)) {
+                return true;
+            } else if (this.searchNearestBlockDelay > 0) {
```

Comment:
Is the number this.searchNearestBlockDelay more likely going to be -1 or any number > 0?

Reply:
Even if `searchNearestBlockDelay` is most likely to be `-1` the comparision could be performed concurrently without possibly changing the program outcome, so I actually expect it to be more performant if we rearrange the blocks regardless of likely valid (though your suspicion that it is more likely to be > 0 is correct).

Reply:
Yeah very small but better if we just swap the blocks.